### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-sameer-pathan**
### Description:
- Introduced new connector with following actions
    - **GET ALERT LIST** :  
        - Retrieves information about alerts generated by Pure1-monitored appliances from Pure Storage.
    - **GET ARRAY LIST**: 
        - Retrieves information about FlashArray and FlashBlade storage appliances from Pure Storage.
    - **GET ARRAY SUPPORT CONTRACTS**: 
        - Retrieves the support contracts associated with arrays from Pure Storage.
    - **GET ARRAY TAGS**: 
         - Retrieves the tags associated with specified arrays from Pure Storage.
    - **DELETE ARRAY TAGS**: 
         - Deletes the tags from Pure Storage based on input parameters specified.
    - **CREATE OR UPDATE ARRAY TAGS**: 
        - Creates or Updates Array Tags Pure Storage based on input parameters specified.
    - **GET CONTROLLER LIST**: 
         - Retrieves information about FlashArray controllers from Pure Storage.
    - **GET DIRECTORY LIST**: 
         - Retrieves information about FlashArray managed directory objects from Pure Storage.
    - **GET FLASHARRAY DRIVE LIST**: 
         - Retrieves information about FlashArray drives from Pure Storage..
#### UTCs:
- [x]  Installed connector using `tgz` file 
- [x] Verified connector logo.
- [x] Verified action and playbook list.
- [x] Verified Action parameters.

### CODE REVIEW

File: pure-storage/info.json [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "tooltip": "Specify the comma-separated alert IDs to retrieve the corresponding alerts. If there is not at least one resource that matches each id element, an error is returned.",
+          "description": "(Optional) Specify the comma-separated alert IDs to retrieve the corresponding alerts. If there is not at least one resource that matches each id element, an error is returned."
+        },
+        {
+          "name": "names",
+          "title": "Names",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: This can be renamed as "Alert Names"

File: pure-storage/info.json [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
+          "description": "(Optional) Specify the filter criteria to retrieve required arrays. Single quotes are required around all strings inside the filters."
+        },
+        {
+          "name": "fqdns",
+          "title": "FQDNs",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "placeholder": "example: apt35,foo,bar",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: Is this a valid placeholder?

File: pure-storage/info.json [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+      "enabled": true,
+      "parameters": [
+        {
+          "name": "resource_ids",
+          "title": "Resource IDs",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "tooltip": "Specify the comma-separated resource IDs to retrieve the corresponding arrays. If there is not at least one resource that matches each id element, an error is returned.",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: Please update the description before sending this to infodev team.

File: pure-storage/info.json [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+          "description": "(Optional) Specify the comma-separated resource IDs to retrieve the corresponding arrays. If there is not at least one resource that matches each id element, an error is returned."
+        },
+        {
+          "name": "resource_names",
+          "title": "Resource Names",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "tooltip": "Specify the comma-separated resource names to retrieve the corresponding arrays. If there is not at least one resource that matches each name element, an error is returned.",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: Please update the description before sending this to infodev team.

File: pure-storage/info.json [hunk 5]
```diff
@@ -0,0 +1,775 @@
...
+          "description": "(Optional) Specify the filter criteria to retrieve required support contracts. Single quotes are required around all strings inside the filters."
+        },
+        {
+          "name": "resource_fqdns",
+          "title": "Resource FQDNs",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "placeholder": "example: apt35,foo,bar",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: Is it a valid placeholder?

File: pure-storage/info.json [hunk 6]
```diff
@@ -0,0 +1,775 @@
...
+      "enabled": true,
+      "parameters": [
+        {
+          "name": "resource_ids",
+          "title": "Resource IDs",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "tooltip": "Specify the comma-separated resource IDs to retrieve the corresponding arrays. If there is not at least one resource that matches each id element, an error is returned.",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: Update Description..

File: pure-storage/info.json [hunk 7]
```diff
@@ -0,0 +1,775 @@
...
+          "type": "json",
+          "editable": true,
+          "visible": true,
+          "required": true,
+          "value": "[\n  {\n    \"key\": \"example_key\",\n    \"value\": \"value123\"\n  }\n]",
+          "tooltip": "Specify list of tags to be upserted.",
+          "description": "Specify list of tags to be upserted. For example [\n  {\n    \"key\": \"example_key\",\n    \"value\": \"value123\"\n  }\n]"
+        },
+        {
+          "name": "resource_ids",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json:481]: can we keep a select here cause any one of the parameter(i.e ID or Name) is req.??

File: pure-storage/info.json [hunk 8]
```diff
@@ -0,0 +1,775 @@
...
+        },
+        {
+          "name": "names",
+          "title": "Controller Names",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "tooltip": "Specify the comma-separated controller names to retrieve the corresponding controllers. If there is not at least one resource that matches each name element, an error is returned.",
+          "description": "(Optional) Specify the comma-separated controller to retrieve the corresponding controllers. If there is not at least one resource that matches each name element, an error is returned."
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: update desc

File: pure-storage/info.json [hunk 9]
```diff
@@ -0,0 +1,775 @@
...
+        },
+           {
+          "name": "file_system_ids",
+          "title": "File System IDs",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": false,
+          "placeholder": "For example file_system_id1, file_system_id2",
+          "tooltip": "Specify the comma-separated directory IDs to retrieve the corresponding directories. If there is not at least one resource that matches each id element, an error is returned.",
```
Comment (cs-harsh-sharma) [inline pure-storage/info.json]: update tooltip

File: pure-storage/info.json (hunk 7 above)
Comment (cs-sameer-pathan) [reply pure-storage/info.json:481]: It is clear that we have to provide at least one of them but it is not clear whether can we provide both parameters at same time or not ?

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**